src/common/shared_cache.hpp

```diff
@@ -105,7 +105,9 @@ class SharedLRU {
       lderr(cct) << "leaked refs:\n";
       dump_weak_refs(*_dout);
       *_dout << dendl;
-      assert(weak_refs.empty());
+      if (cct->_conf->get_val<bool>("debug_asserts_on_shutdown")) {
+	assert(weak_refs.empty());
```

Comment:
Looks like something went wrong here and it used a tab instead of spaces.

Reply:
Guess this was in the original. Not quite sure how that happened?

Reply:
It seems it is a  methods for its avoidance. 
But do you know the reasons why assert during shuting down? I cannot reproduce this bug.

Reply:
I don't remember for certain what prompted this one. There have been varying causes over the years, generally found in testing rather than by users and squashed as soon as we identified them. But we decided anything the users ran into first we didn't want to have assert out when they were shutting stuff down anyway.

Reply:
I agree with you . Shutting down when assert may cause some unpredictable problems.
We could add some log to indicate it before skipping assert.